testcases/kernel/fs/linktest/linktest.sh

```diff
@@ -79,7 +79,7 @@ do_link() {
 		RTYPE=TFAIL
 	fi
 
-	tst_res $RTYPE "" "$prefix_msg Link Errors: $lerrors"
+	tst_resm $RTYPE "$prefix_msg Link Errors: $lerrors"
 
 	: $(( TST_COUNT += 1 ))
 
```

Comment:
Good catch. I've modified the patch to use the test.sh library so that the failure is actually propagated as the test exit value and also made use of ```tst_tmpdir``` and ```tst_rmdir``` and pushed, thanks.